futures-util/src/io/framed_write.rs

```diff
@@ -170,6 +167,33 @@ impl<T> FramedWrite2<T> {
     }
 }
 
+impl<T: AsyncWrite + Encoder> FramedWrite2<T> {
+    fn poll_internal(&mut self, cx: &mut task::Context, do_close: bool) -> Poll<(), T::Error> {
+        trace!("flushing framed transport");
+
+        while !self.buffer.is_empty() {
+            trace!("writing; remaining={}", self.buffer.len());
+
+            let n = try_ready!(self.inner.poll_write(cx, &self.buffer));
+
+            if n == 0 {
+                return Err(io::Error::new(io::ErrorKind::WriteZero, "failed to
+                                          write frame to transport").into());
+            }
+
+            // TODO: Add a way to `bytes` to do this w/o returning the drained
+            // data.
+            let _ = self.buffer.split_to(n);
+        }
+
+        if do_close {
```

Comment:
seems like this could just be inlined into the callers to get rid of the flag

Reply:
Done